### ISSUE TITLE
Help from tslint-eslint rules?

### ISSUE THREAD

**Issue by gaurav21r** [eslint/typescript-eslint-parser#74]
Hi there! I don't know a lot about the TS Language Services API but I am cheering for ESLint to support TS very soon! ESLint is pretty fantastic and I think we would all win if there are fewer tools covering a larger area! Besides, eslint configs are far superior to TSLint's.

This project -> https://github.com/buzinas/tslint-eslint-rules uses the reverse methodology. I was just wondering if the core team could have a look at it and get any help from it?

Sorry if this is the wrong place to post this!

**Comment by nzakas** [comment]
@gaurav21r unfortunately, there's not a lot of value to ESLint TypeScript support from looking at that project. Basically, that project will be irrelevant if we do our job correctly because the core ESLint rules will just work. :)

I'm also privately working on a plugin with TypeScript-specific rules: https://github.com/nzakas/eslint-plugin-typescript. That's an area where we could definitely use help (the core ESLint team is not involved in that).

**Comment by gaurav21r** [comment]
@nzakas Thanks for your response! What is the difference between `eslint-plugin-typescript` and `typescript-eslint-parser`? 

If I understood the goals of this project correctly (#13), `typescript-eslint-parser` will enable us to :

1. Use standard ESLint rules (say the airbnb shareable config) with TS Code.
2. Add TS specific rules for TS Code. (Eg: Type annotations should have a space)

**Comment by platinumazure** [comment]
@gaurav21r `typescript-eslint-parser` is a parser that wraps the TypeScript compiler, and it basically transforms TypeScript code into a format that ESLint can use to walk through the code and allow the rules to run on the code. Using just typescript-eslint-parser will allow core ESLint rules to be run on TypeScript code, but any rules specific to TypeScript (e.g., anything that has to do with type inference) will not be run, because ESLint won't allow those rules into the core rule set (due to not being widely applicable to all JavaScript).

`eslint-plugin-typescript` is an ESLint plugin that contains TypeScript-specific rules. In theory, it could be run on JavaScript code using a parser that isn't typescript-eslint-parser, but in reality the rules would probably crash because they would inspect decorator or type information that the parser simply wouldn't provide.

So basically, the two projects work to add TypeScript to two different steps of the ESLint process-- parsing/code input, and linting rules. Basically, typescript-eslint-parser helps with your first point, and eslint-plugin-typescript will help with your second point.

Make sense?

**Comment by gaurav21r** [comment]
@platinumazure Thanks a lot! Thats a great explanation! I was almost there except I thought this repo would handle both 1 & 2 points. Just to re-iterate, if I set **.eslintrc** to
```js
{
  "extends": "airbnb",
  "parser": "typescript-eslint-parser"
}
```

My typescript code would get linted with the AirBnb rules config?

**Comment by platinumazure** [comment]
Yes- when this project is ready, of course. :-)
